pymc/distributions/moments/means.py

```diff
@@ -0,0 +1,468 @@
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+"""Mean dispatcher for pymc random variables."""
+
+from functools import singledispatch
+
+import numpy as np
+
+from pytensor import tensor as pt
+from pytensor.tensor.math import tanh
+from pytensor.tensor.random.basic import (
+    BernoulliRV,
+    BetaBinomialRV,
+    BetaRV,
+    BinomialRV,
+    CategoricalRV,
+    CauchyRV,
+    DirichletRV,
+    ExponentialRV,
+    GammaRV,
+    GeometricRV,
+    GumbelRV,
+    HalfCauchyRV,
+    HalfNormalRV,
+    HyperGeometricRV,
+    InvGammaRV,
+    LaplaceRV,
+    LogisticRV,
+    LogNormalRV,
+    MultinomialRV,
+    MvNormalRV,
+    NegBinomialRV,
+    NormalRV,
+    ParetoRV,
+    PoissonRV,
+    StudentTRV,
+    TriangularRV,
+    UniformRV,
+    VonMisesRV,
+)
+from pytensor.tensor.var import TensorVariable
+
+from pymc.distributions.continuous import (
+    AsymmetricLaplaceRV,
+    BetaClippedRV,
+    ExGaussianRV,
+    FlatRV,
+    HalfFlatRV,
+    HalfStudentTRV,
+    KumaraswamyRV,
+    LogitNormalRV,
+    MoyalRV,
+    PolyaGammaRV,
+    RiceRV,
+    SkewNormalRV,
+    SkewStudentTRV,
+    WaldRV,
+    WeibullBetaRV,
+)
+from pymc.distributions.discrete import DiscreteUniformRV
+from pymc.distributions.distribution import DiracDeltaRV
+from pymc.distributions.mixture import MarginalMixtureRV
+from pymc.distributions.moments.utils import UndefinedMomentException
+from pymc.distributions.multivariate import (
+    CARRV,
+    DirichletMultinomialRV,
+    KroneckerNormalRV,
+    LKJCorrRV,
+    MatrixNormalRV,
+    MvStudentTRV,
+    StickBreakingWeightsRV,
+    _LKJCholeskyCovRV,
+)
+from pymc.distributions.shape_utils import rv_size_is_none
+
+__all__ = ["mean"]
+
+
+@singledispatch
+def _mean(op, rv, *rv_inputs) -> TensorVariable:
+    raise NotImplementedError(f"Variable {rv} of type {op} has no mean implementation.")
+
+
+def mean(rv: TensorVariable) -> TensorVariable:
+    """Compute the expected value of a random variable.
+
+    The only parameter to this function is the RandomVariable
+    for which the value is to be derived.
+    """
+    return _mean(rv.owner.op, rv, *rv.owner.inputs)
+
+
+def maybe_resize(a: TensorVariable, size) -> TensorVariable:
+    if not rv_size_is_none(size):
+        a = pt.full(size, a)
+    return a
+
+
+@_mean.register(BernoulliRV)
+def bernoulli_mean(op, rv, rng, size, p):
+    return maybe_resize(p, size)
+
+
+@_mean.register(BetaBinomialRV)
+def betabinomial_mean(op, rv, rng, size, n, alpha, beta):
+    return maybe_resize((n * alpha) / (alpha + beta), size)
+
+
+@_mean.register(BetaClippedRV)
+def beta_clipped_mean(op, rv, rng, size, alpha, beta):
+    return maybe_resize(alpha / (alpha + beta), size)
```

Comment:
codecov says this one is untested

Reply:
Also you can dispatch on the base Beta Op class

Reply:
I'll unify the dispatches for both RV Ops. I don't know why we still have the clipped one though.